### PR TITLE
Add Seek::seek_relative

### PR DESCRIPTION
**Pull request by fintelia**
The `BufReader` struct has a `seek_relative` method because its `Seek::seek` implementation involved dumping the internal buffer (https://github.com/rust-lang/rust/issues/31100).

Unfortunately, there isn't really a good way to take advantage of that method in generic code. This PR adds the same method to the main `Seek` trait with the straightforward default method, and an override for `BufReader` that calls its implementation.

_Also discussed in [this](https://internals.rust-lang.org/t/add-seek-seek-relative/19546) internals.rust-lang.org thread._

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
New unstable APIs need an approved [API change proposal](https://github.com/rust-lang/libs-team/issues/new?assignees=&labels=api-change-proposal%2C+T-libs-api&projects=&template=api-change-proposal.md&title=%28My+API+Change+Proposal%29) - can you file one and then poke here again once/if that's approved?

@rustbot author

File: library/std/src/io/mod.rs [hunk 1]
```diff
@@ -1957,6 +1957,32 @@ pub trait Seek {
...
+    /// };
+    ///
+    /// fn main() -> io::Result<()> {
+    ///     let mut f = File::open("foo.txt")?;
+    ///     f.seek_relative(10)?;
+    ///     assert_eq!(f.stream_position()?, 10);
+    ///     Ok(())
+    /// }
+    /// ```
+    #[unstable(feature = "seek_seek_relative", issue = "none")]
```
Comment (m-ou-se) [inline library/std/src/io/mod.rs:1981]: Can you [open a tracking issue](https://github.com/rust-lang/rust/issues/new?assignees=&labels=C-tracking-issue%2C+T-libs&template=library_tracking_issue.md&title=Tracking+Issue+for+XXX) for this? Thanks!

Comment (fintelia) [reply library/std/src/io/mod.rs]: Done!

File: library/std/src/io/mod.rs [hunk 2]
```diff
@@ -1957,6 +1957,32 @@ pub trait Seek {
     fn stream_position(&mut self) -> Result<u64> {
         self.seek(SeekFrom::Current(0))
     }
+
+    /// Seeks relative to the current position.
+    ///
+    /// This is equivalent to `self.seek(SeekFrom::Current(offset))`.
```
Comment (the8472) [inline library/std/src/io/mod.rs:1963]: The documentation could be improved.

It should say that the default implementation is equivalent to that but doesn't return the new position which can allow some implementations such as `BufWriter` (+ link to bufwriter trait impl docs) to perform more efficient seeks.

**Comment by fintelia** [comment]
I think I've addressed all the concerns that have been raised?

**Pull request merged**